test/UnitTests/Microsoft.Testing.Platform.UnitTests/Services/TestApplicationResultTests.cs

```diff
@@ -68,15 +68,20 @@ public async Task GetProcessExitCodeAsync_If_Failed_Tests_Returns_AtLeastOneTest
     public async Task GetProcessExitCodeAsync_If_Canceled_Returns_TestSessionAborted()
     {
         Mock<ITestApplicationCancellationTokenSource> testApplicationCancellationTokenSource = new();
+        var policiesService = new PoliciesService();
         testApplicationCancellationTokenSource.SetupGet(x => x.CancellationToken).Returns(() =>
         {
             CancellationTokenSource cancellationTokenSource = new();
+
+            // In reality, this is done by CTRLPlusCCancellationTokenSource (the real production impl of ITestApplicationCancellationTokenSource).
+            cancellationTokenSource.Token.Register(() => policiesService.AbortPolicy.ExecuteCallbacksAsync(cancellationToken: default).GetAwaiter().GetResult());
```

Comment:
```suggestion
            cancellationTokenSource.Token.Register(async () => await policiesService.AbortPolicy.ExecuteCallbacksAsync(cancellationToken: default));
```

Reply:
@Evangelink That will be fire-and-forget. CancellationToken.Register doesn't have a `Func<Task>` overload 😕